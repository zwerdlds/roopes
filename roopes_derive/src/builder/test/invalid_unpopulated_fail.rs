use roopes::prelude::*;

#[derive(Builder)]
struct TestStruct {
    value_one: i32,
}

fn main(){
    let builder = TestStructBuilder::new();

    let test_struct = builder.build();

    assert_eq!(0, test_struct.value_one);
}