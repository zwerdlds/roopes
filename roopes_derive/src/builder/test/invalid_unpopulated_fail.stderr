error[E0599]: no method named `build` found for struct `TestStructBuilder<FieldValueOneUnset>` in the current scope
  --> src/builder/test/invalid_unpopulated_fail.rs:11:31
   |
3  | #[derive(Builder)]
   |          ------- method `build` not found for this struct
...
11 |     let test_struct = builder.build();
   |                               ^^^^^ method not found in `TestStructBuilder<FieldValueOneUnset>`
   |
   = note: the method was found for
           - `TestStructBuilder<FieldValueOneSet>`
